### ISSUE TITLE
Enabling org-roam leads to launch error finalizer failed: (wrong-type-argument sqlitep nil)

### ISSUE THREAD

**Issue by Jdogzz** [doomemacs/doomemacs#8066]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

Launching emacs has no errors.

### Current behavior

The `*Messages*` buffer shows `finalizer failed: (wrong-type-argument sqlitep nil)`. As far as I can tell there is no overt functionality problem.

There are at least two mentions of this problem online with no resolution:

https://github.com/seagle0128/.emacs.d/issues/399

https://org-roam.discourse.group/t/what-is-responsible-for-this-db-sync-finalizer-failure-message/3567/1

### Steps to reproduce

1. Enable roam v2.
2. Open emacs.
3. Check `*Messages`* buffer.

### System Information

https://pastebin.com/4DMfJsXh

**Comment by hlissner** [comment]
I'm not sure what to make of this, since I cannot reproduce it. The one difference between our setups is `org-roam-ui`. Could you try disabling it and see if that has any effect?

**Comment by Jdogzz** [comment]
Thanks for looking into this. I disabled `org-roam-ui`, did a sync, and reopened, but the error still crops up. However, I have good news! I started removing parts of my org-roam config and it appears the line in my config causing error on emacs startup is:

```
(org-roam-db-autosync-mode)
```

I removed that line since I see the roam v2 module already takes care of getting orgroam set up:

https://github.com/doomemacs/doomemacs/blob/master/modules/lang/org/contrib/roam2.el#L47

After that, doing some orgroam stuff, searching nodes, adding notes, etc., I do not see this error in Messages anymore. Since I see a couple of thumbs up on this issue, if the same line was in your config, does removing it and trying to do some orgroam actions likewise stop the error from happening? I'm happy to close this issue if so.

**Comment by j-palms** [comment]
No, not for me. I completely removed all org-roam related parts from my config and this error always gets logged after startup.

**Comment by Lildragonxxl** [comment]
Same problem as you encountered, but even after removing `(org-roam-db-autosync-mode)`, the same error occurs after running `M-x org-roam-db-sync`, as follows:

`Clearing removed files...done
Processing modified files...done
finalizer failed: (wrong-type-argument sqlitep nil)`

However, I've checked that this reported error doesn't affect my ability to create notes and view linked nodes, so I don't know if this is a bug or if it affects other aspects of use.

My version of emacs is emacs-plus version 29.4, org-roam v2, and I'm using macOS.However, I seem to have found that it may be that the org-roam is not linking correctly to sqlite related functions, but I don't know much about this and so far I haven't been able to fix it, so if anyone has solved it please also leave a tutorial 😂

**Comment by adithyabhatkajake** [comment]
I had the same error. The message disappeared after I removed the `(org-roam-db-autosync-mode)` line from the `use-package!` declaration.

**Comment by Lildragonxxl** [comment]
> I had the same error. The message disappeared after I removed the `(org-roam-db-autosync-mode)` line from the `use-package!` declaration.

It may happen again if you run `M-x org-roam-db-sync' manually. Have you ever tried this?

**Comment by christianromney** [comment]
Can confirm this happens on emacs-plus 30.1.

**Comment by FrederickGeek8** [comment]
I have been experiencing this for a while, but the [org-roam discourse link in the OP](https://org-roam.discourse.group/t/what-is-responsible-for-this-db-sync-finalizer-failure-message/3567/1) has a solution that works for me. [This diagnosis](https://org-roam.discourse.group/t/what-is-responsible-for-this-db-sync-finalizer-failure-message/3567/12) aligns with what I saw with `(setq debug-on-signal t)` and [their later solution](https://org-roam.discourse.group/t/what-is-responsible-for-this-db-sync-finalizer-failure-message/3567/13) also works out.

I'm copy-pasting the [solution](https://org-roam.discourse.group/t/what-is-responsible-for-this-db-sync-finalizer-failure-message/3567/13) below, but I would encourage people to check it out:
```elisp
(defun patch/emacsql-close (connection &rest args)
  "Prevent calling emacsql-close if connection handle is nil."
  (when (oref connection handle)
    t))

(advice-add 'emacsql-close :before-while #'patch/emacsql-close)
```

I'm not sure if this is related to Doom Emacs, but several people in that thread _were using_ Doom Emacs.

For the record, this is the stacktrace I got when I set `debug-on-signal`, and while there is a bunch of pointer garbage, you can see the duplicated `sqlite-close` that is described in the [discourse analysis](https://org-roam.discourse.group/t/what-is-responsible-for-this-db-sync-finalizer-failure-message/3567/12):
```
Debugger entered--Lisp error: (wrong-type-argument sqlitep nil)
  sqlite-close(nil)
  #f(compiled-function (connection) #<bytecode 0x160223ee2a330fe8>)(#<emacsql-sqlite-builtin-connection emacsql-sqlite-builtin-connection-1fee517d6ef6>)
  apply(#f(compiled-function (connection) #<bytecode 0x160223ee2a330fe8>) #<emacsql-sqlite-builtin-connection emacsql-sqlite-builtin-connection-1fee517d6ef6> nil)
  emacsql-close(#<emacsql-sqlite-builtin-connection emacsql-sqlite-builtin-connection-1fee517d6ef6>)
  #f(compiled-function () #<bytecode 0xdac16f5f11089b5>)()
  garbage-collect()
  gcmh-idle-garbage-collect()
  apply(gcmh-idle-garbage-collect nil)
  timer-event-handler([t 26640 59779 664487 nil gcmh-idle-garbage-collect nil nil 999999 nil])
```
